### ISSUE TITLE
shelly dimmer 2 not dimmable

### ISSUE THREAD

**Issue by smhc** [Luligu/matterbridge-shelly#65]
This was working previously, but appears to have broken around version 1.0.0 or greater.

An error in the logs states:

:02:29.347] [InteractionServer]: Validation-Error 0x80 while invoking command: (ValidationMandatoryFieldMissingError/128) Missing mandatory field transitionTime in field transitionTime

**Comment by Luligu** [comment]
Hi,
without more context, I would say that the error is generated from the controller (what controller are you using?) sending a command that is not compliant.

When you dimmer you should see something like this:
![image](https://github.com/user-attachments/assets/b0c1cd42-8484-4919-a9c7-2f96fbe22e35)

So the issue is not generated by matterbridge or shelly plugin.

If you want more info I need the full log of that part when your controller sends the command to matter.

My best.

**Comment by smhc** [comment]
I am using google home controller, with the Android google "home app". It was working ok previously.

I have captured the matter logs below:


```
[22:13:35.105] [SubscriptionHandler] [debug] Sending subscription update message for ID 1759468559 with 0 attributes and 0 events
[22:13:35.107] [MessageExchange] [debug] New exchange channel: udp://fe80::2680:4e2f:ee1:11f7%11:5540 on session secure/2453 protocol: 1 id: 23542 session: secure/2453 peerSessionId: 35253 active threshold ms: 4000 active interval ms: 300 idle interval ms: 500 maxTransmissions: 5 useMrp: true
[22:13:35.109] [ExchangeManager] [debug] Message » id: 35253/23542/211084599 type: 1/5 acked: undefined reqAck: true duplicate: false payload: 1526000f5cdf68290424ff0c18
[22:13:35.107] [SubscriptionHandler] [debug] Sending subscription changes for ID 1759468559:
[22:13:35.109] [InteractionMessenger] [debug] Sending DataReport chunk with 0 attributes and 0 events: 13 bytes (moreChunkedMessages: false, suppressResponse: true)
[22:13:35.132] [MessageExchange] [debug] Message « id: 2453/23542/31452435 type: 0/16 acked: 211084599 reqAck: false duplicate: false payload: 
[22:13:41.922] [MessageExchange] [debug] New exchange channel: udp://fe80::2680:4e2f:ee1:11f7%11:5540 on session secure/2453 protocol: 1 id: 55950 session: secure/2453 peerSessionId: 2453 active threshold ms: 4000 active interval ms: 300 idle interval ms: 500 maxTransmissions: 5 useMrp: true
[22:13:41.922] [MessageExchange] [debug] Message « id: 2453/55950/31452436 type: 1/8 acked: undefined reqAck: true duplicate: false payload: 1528002801360215370024002624010824020418350124007518181824ff0b18
[22:13:41.925] [ExchangeManager] [debug] Message » id: 2453/55950/211084600 type: 1/9 acked: 31452436 reqAck: true duplicate: false payload: 152800360115350137002400262401082402041835012400851818181824ff0c18
[22:13:41.923] [InteractionServer] [debug] Received invoke request from udp://fe80::2680:4e2f:ee1:11f7%11:5540 on session secure/2453 with 1 commands: MA-dimmablelight(0x26)/LevelControl(0x8)/moveToLevelWithOnOff(0x4), suppressResponse=false
[22:13:41.924] [InteractionServer] [info] Validation-Error 0x80 while invoking command: (ValidationMandatoryFieldMissingError/128) Missing mandatory field transitionTime in field transitionTime
[22:13:41.934] [MessageExchange] [debug] Message « id: 2453/55950/31452437 type: 0/16 acked: 211084600 reqAck: false duplicate: false payload: 
[22:14:00.897] [SubscriptionHandler] [debug] Sending subscription update message for ID 1759468559 with 0 attributes and 0 events
[22:14:00.897] [MessageExchange] [debug] New exchange channel: udp://fe80::2680:4e2f:ee1:11f7%11:5540 on session secure/2453 protocol: 1 id: 23543 session: secure/2453 peerSessionId: 35253 active threshold ms: 4000 active interval ms: 300 idle interval ms: 500 maxTransmissions: 5 useMrp: true
[22:14:00.898] [SubscriptionHandler] [debug] Sending subscription changes for ID 1759468559:
[22:14:00.899] [ExchangeManager] [debug] Message » id: 35253/23543/211084601 type: 1/5 acked: undefined reqAck: true duplicate: false payload: 1526000f5cdf68290424ff0c18
[22:14:00.898] [InteractionMessenger] [debug] Sending DataReport chunk with 0 attributes and 0 events: 13 bytes (moreChunkedMessages: false, suppressResponse: true)
[22:14:00.908] [MessageExchange] [debug] Message « id: 2453/23543/31452438 type: 0/16 acked: 211084601 reqAck: false duplicate: false payload: 
```

I could try removing and recommissioning the bridge.

**Comment by Luligu** [comment]
For the log follow this please:
https://github.com/Luligu/matterbridge/blob/main/README.md#how-to-send-the-debug-log-files

The constraint on Matter specs are:
![image](https://github.com/user-attachments/assets/3b8abf92-89bd-4b9c-b929-523e81e46254)

From the log made like I asked you, we can understand at least what is the controller sending...

**Comment by smhc** [comment]
[matter (1).log](https://github.com/user-attachments/files/17395801/matter.1.log)
[matterbridge (1).log](https://github.com/user-attachments/files/17395802/matterbridge.1.log)

I've attached the logs for the scenario with just the dimmer in the whitelist. From what I can see the logs don't show much more than what I've posted.

**Comment by Luligu** [comment]
Sorry you are right we need the console log like in the screenshot I posted.

**Comment by smhc** [comment]
I've captured the logs from the console with matterbridge.cmd --matterlogger debug --logger debug
but it still doesn't show too much.  I've also attached the shelly dimmer config capture.

[shellydimmer2-EC64C9C669A1.json](https://github.com/user-attachments/files/17402579/shellydimmer2-EC64C9C669A1.json)


```
[06:52:01.012] [MessageExchange] New exchange channel: udp://fd06:8df3:e47f:940:47c:8041:fa75:3c56:5540 on session secure/60275 protocol: 1 id: 52855 session: secure/60275 peerSessionId: 62109 active threshold ms: 4000 active interval ms: 300 idle interval ms: 500 maxTransmissions: 5 useMrp: true
[06:52:01.012] [SubscriptionHandler] Sending subscription changes for ID 2633879218: MA-onoffswitch(0x23)/OnOff(0x6)/onOff(0x0)=true (3388713697)
[06:52:01.012] [InteractionMessenger] Sending DataReport chunk with 1 attributes and 0 events: 41 bytes (moreChunkedMessages: false, suppressResponse: false)
[06:52:01.012] [ExchangeManager] Message » id: 62109/52855/250150751 type: 1/5 acked: undefined reqAck: true duplicate: false payload: 152600b2cefd9c36011535012600e1aafbc93701240223240306240400182902181818280424ff0c18
[06:52:01.013] [SubscriptionHandler] Sending subscription update message for ID 2633879219 with 1 attributes and 0 events
[06:52:01.013] [MessageExchange] New exchange channel: udp://fe80::2680:4e2f:ee1:11f7%11:5540 on session secure/60276 protocol: 1 id: 52856 session: secure/60276 peerSessionId: 16770 active threshold ms: 4000 active interval ms: 300 idle interval ms: 500 maxTransmissions: 5 useMrp: true
[06:52:01.013] [SubscriptionHandler] Sending subscription changes for ID 2633879219: MA-onoffswitch(0x23)/OnOff(0x6)/onOff(0x0)=true (3388713697)
[06:52:01.014] [InteractionMessenger] Sending DataReport chunk with 1 attributes and 0 events: 41 bytes (moreChunkedMessages: false, suppressResponse: false)
[06:52:01.014] [ExchangeManager] Message » id: 16770/52856/113228797 type: 1/5 acked: undefined reqAck: true duplicate: false payload: 152600b3cefd9c36011535012600e1aafbc93701240223240306240400182902181818280424ff0c18
[06:52:01.017] [MessageExchange] Message « id: 60275/52855/172053323 type: 1/1 acked: 250150751 reqAck: true duplicate: false payload: 1524000024ff0b18
[06:52:01.017] [ExchangeManager] Message » id: 62109/52855/250150752 type: 0/16 acked: 172053323 reqAck: false duplicate: false payload:
[06:52:01.026] [MessageExchange] Message « id: 60276/52856/248939774 type: 1/1 acked: 113228797 reqAck: true duplicate: false payload: 1524000024ff0b18
[06:52:01.027] [ExchangeManager] Message » id: 16770/52856/113228798 type: 0/16 acked: 248939774 reqAck: false duplicate: false payload:
[06:52:02.465] [MatterDevice] Announcing fabric: 13127518232343345540
[06:52:02.478] [MdnsBroadcaster] Announcement Generator: Fabric id: 950937b1b373e635/2247942442 qname: 950937B1B373E635-0000000085FCE12A._matter._tcp.local port: 5540 interface: Tailscale
[06:52:02.478] [MdnsBroadcaster] Announcement Generator: Fabric id: 950937b1b373e635/2247942442 qname: 950937B1B373E635-0000000085FCE12A._matter._tcp.local port: 5540 interface: Ethernet
[06:52:26.535] [MessageExchange] New exchange channel: udp://fe80::2680:4e2f:ee1:11f7%11:5540 on session secure/60276 protocol: 1 id: 46805 session: secure/60276 peerSessionId: 60276 active threshold ms: 4000 active interval ms: 300 idle interval ms: 500 maxTransmissions: 5 useMrp: true
[06:52:26.535] [MessageExchange] Message « id: 60276/46805/248939775 type: 1/8 acked: undefined reqAck: true duplicate: false payload: 1528002801360215370024002624010824020418350124005918181824ff0b18
[06:52:26.536] [InteractionServer] Received invoke request from udp://fe80::2680:4e2f:ee1:11f7%11:5540 on session secure/60276 with 1 commands: MA-dimmablelight(0x26)/LevelControl(0x8)/moveToLevelWithOnOff(0x4), suppressResponse=false
[06:52:26.538] [InteractionServer] Validation-Error 0x80 while invoking command: (ValidationMandatoryFieldMissingError/128) Missing mandatory field transitionTime in field transitionTime
[06:52:26.539] [ExchangeManager] Message » id: 60276/46805/113228799 type: 1/9 acked: 248939775 reqAck: true duplicate: false payload: 152800360115350137002400262401082402041835012400851818181824ff0c18
[06:52:26.548] [MessageExchange] Message « id: 60276/46805/248939776 type: 0/16 acked: 113228799 reqAck: false duplicate: false payload:
[06:52:30.235] [SubscriptionHandler] Sending subscription update message for ID 2633879219 with 0 attributes and 0 events
[06:52:30.236] [MessageExchange] New exchange channel: udp://fe80::2680:4e2f:ee1:11f7%11:5540 on session secure/60276 protocol: 1 id: 52857 session: secure/60276 peerSessionId: 16770 active threshold ms: 4000 active interval ms: 300 idle interval ms: 500 maxTransmissions: 5 useMrp: true
[06:52:30.236] [SubscriptionHandler] Sending subscription changes for ID 2633879219:
[06:52:30.237] [InteractionMessenger] Sending DataReport chunk with 0 attributes and 0 events: 13 bytes (moreChunkedMessages: false, suppressResponse: true
[06:52:30.237] [ExchangeManager] Message » id: 16770/52857/113228800 type: 1/5 acked: undefined reqAck: true duplicate: false payload: 152600b3cefd9c290424ff0c18
[06:52:30.247] [MessageExchange] Message « id: 60276/52857/248939777 type: 0/16 acked: 113228800 reqAck: false duplicate: false payload:
[06:52:31.264] [SubscriptionHandler] Sending subscription update message for ID 2633879218 with 0 attributes and 0 events
[06:52:31.264] [MessageExchange] New exchange channel: udp://fd06:8df3:e47f:940:47c:8041:fa75:3c56:5540 on session secure/60275 protocol: 1 id: 52858 session: secure/60275 peerSessionId: 62109 active threshold ms: 4000 active interval ms: 300 idle interval ms: 500 maxTransmissions: 5 useMrp: true
[06:52:31.264] [SubscriptionHandler] Sending subscription changes for ID 2633879218:
[06:52:31.265] [InteractionMessenger] Sending DataReport chunk with 0 attributes and 0 events: 13 bytes (moreChunkedMessages: false, suppressResponse: true
[06:52:31.265] [ExchangeManager] Message » id: 62109/52858/250150753 type: 1/5 acked: undefined reqAck: true duplicate: false payload: 152600b2cefd9c290424ff0c18
[06:52:31.269] [MessageExchange] Message « id: 60275/52858/172053324 type: 0/16 acked: 250150753 reqAck: false duplicate: false payload:
[06:52:38.867] [Kitchen Pendant Lights] WebSocket update from device shelly1pmminig3-34B7DA908E68 host 192.168.1.186
[06:52:38.868] [Kitchen Pendant Lights] switch:0:current updated from 0.228 to 0.227 in component switch:0 (Switch)
[06:52:38.868] [Kitchen Pendant Lights] Shelly message for device shelly1pmminig3-34B7DA908E68 Switch:switch:0:current:0.227
```

**Comment by smhc** [comment]
I've reconfigured my hub to leave the public preview programme - it's possibly an issue with the 20.20240530.3.18053 firmware. Will try again after it eventually rolls back.

**Comment by smhc** [comment]
I turned off the nest hub and only had the nest mini's running and lights were able to dim ok. Presumably it's a bug in the nest hub preview programme firmware. Hopefully it's not a change they plan to roll out to the production releases.

I'll stay on the public release rather than preview release train for the devices and apps from now on.

**Comment by smhc** [comment]
FYI I've had to resolve this with a hack in matter.js as the nest hub would not roll back to production firmware. I had to change LevelControlCluster.js from:

```js
  LevelControl2.TlvMoveToLevelRequest = TlvObject({
    level: TlvField(0, TlvUInt8.bound({ max: 254 })),
    transitionTime: TlvField(1, TlvNullable(TlvUInt16)),
    optionsMask: TlvField(2, TlvBitmap(TlvUInt8, LevelControl2.Options)),
    optionsOverride: TlvField(3, TlvBitmap(TlvUInt8, LevelControl2.Options))
  });
```

to this

```js
  LevelControl2.TlvMoveToLevelRequest = TlvObject({
    level: TlvField(0, TlvUInt8.bound({ max: 254 })),
    transitionTime: TlvOptionalField(1, TlvNullable(TlvUInt16)),
    optionsMask: TlvOptionalField(2, TlvBitmap(TlvUInt8, LevelControl2.Options)),
    optionsOverride: TlvOptionalField(3, TlvBitmap(TlvUInt8, LevelControl2.Options))
  });
```

I'm not sure if its a bug with matter.js or with the google controller. But I do believe the options fields are supposed to be optional.

**Comment by smhc** [comment]
After closer inspection, I think it might be this bug in matter.js: https://github.com/project-chip/matter.js/pull/848

**Comment by Luligu** [comment]
Hi,
I understood you solved the issue since you closed...
Did you open an issue in matter.js ?

**Comment by Apollon77** [comment]
Thanks for bringing it to our attention with https://github.com/project-chip/matter.js/pull/1332 ... but please report to Google! They need to fix this not compliant behavior

**Comment by smhc** [comment]
As noted the nest hub would not update from the preview to production firmware so I had to hack in a temporary workaround. I have simply documented the workaround here for the paper trail and to help others.

**Comment by Apollon77** [comment]
@smhc The definition of the "default" values is mainly that way for attributes, but an mandatory field should be a mandatory field still, so basically from the current standpoint the implementation in matter.js seems correct.

**Comment by smhc** [comment]
Unfortunately this seems broken again with nest firmware 22.20240805.103.161, matterbridge 1.6.5 and matterbridge-shelly 1.0.11.

There isn't anything obvious in the logs like last time, I'll see if I can capture something.

**Comment by Apollon77** [comment]
Can you please post more log? like the invoke message itself and a bit afterwards? There could also be a chance of not being google related

**Comment by smhc** [comment]
There is nothing logged when performing the dim operation. The failure appears to occur on the client end with no messages sent. I theorise it may be a misconfiguration of the registration of the device itself. I tried "rediscovering" the device via mdns but it did not resolve the issue. This is the config listed in matterbridge:

```
10 (light:0)	LevelControl	0x08	currentLevel	0x00	254
10 (light:0)	LevelControl	0x08	maxLevel	0x03	254
10 (light:0)	LevelControl	0x08	options	0x0f	{ executeIfOff: false, coupleColorTempToLevel: false }
10 (light:0)	LevelControl	0x08	onLevel	0x11	null
10 (light:0)	LevelControl	0x08	clusterRevision	0xfffd	5
10 (light:0)	LevelControl	0x08	featureMap	0xfffc	{ onOff: true, lighting: false, frequency: false }
10 (light:0)	LevelControl	0x08	attributeList	0xfffb	[ 0, 2, 3, 15, 17, 65528, 65529, 65531, 65532, 65533 ]
10 (light:0)	LevelControl	0x08	acceptedCommandList	0xfff9	[ 0, 1, 2, 3, 4, 5, 6, 7 ]
10 (light:0)	LevelControl	0x08	generatedCommandList	0xfff8	[ ]
10 (light:0)	LevelControl	0x08	minLevel	0x02	0
```


The "lighting: false" looks a bit suspicious.

**Comment by smhc** [comment]
Any luck reproducing this, or did you want my shelly config to look at?
Perhaps it's related to the fact I have the shelly configured with a "one button mode" button type?

I can raise a new issue if you think this bug is legitimate instead of tracking it under this closed issue.

**Comment by Luligu** [comment]
I updated the cluster but I will publish it like dev this late evening cause I’m traveling right now..
I don’t think it will solve but we will see… in any case it had to be done with a few more stuff to be compliant 100 with Matter 1.3 specs since we will start certification in January probably.
Thanks no need to open a new one.

**Comment by Luligu** [comment]
Try with 1.6.6 if it is solved…

**Comment by smhc** [comment]
Thanks, this seems to have resolved it. I did need to reset and recommission, but perhaps it would have resolved itself if I waited long enough.

**Comment by Luligu** [comment]
So you think was the cluster not fully compliant with the device type?
If it is this way is a really interesting discovery…

**Comment by smhc** [comment]
The 'lighting: true' took effect immediately, but the controllers (I guess?) weren't working with it until I recommissioned. As I said, they may have worked if I just waited long enough.

As for why it was an issue in the first place, I don't know. Probably just an effect of loading config from storage that was incompatible with newer versions of matterbridge/shelly. My setup itself isn't very complicated, it's just matterbridge-shelly and some stock shelly PMs and one stock shelly dimmer 2.
